network.go

```diff
@@ -52,6 +52,8 @@ type NetworkController interface {
 	// Create a new network. The options parameter carry driver specific options.
 	// Labels support will be added in the near future.
 	NewNetwork(d *NetworkDriver, name string, options interface{}) (Network, error)
+	// Register an external driver.
+	RegisterDriver(name string, driver driverapi.Driver)
```

Comment:
This has no dependency on how plugins work. That would be the point of doing it this way: to insulate against decisions made elsewhere about how plugins work.